File: tide-canvas-go/internal/model/notification.go
package model

// 通知类型（type 列，varchar）。
const (
	NotificationTypeFollow  = "follow"  // 关注
	NotificationTypeComment = "comment" // 评论
	NotificationTypeLike    = "like"    // 点赞
	NotificationTypeTip     = "tip"     // 博客打赏
)

// 通知目标类型（target_type 列，varchar；空串表示无具体目标，如关注通知）。
const (
	NotificationTargetPost = "post" // 社区帖子
	NotificationTargetBlog = "blog" // 博客
)

// SysNotification 站内通知表 sys_notification（中间/流水表，内部:无 public_id）。
//
// 用 BaseModel（雪花主键 + create_time/update_time，无逻辑删除）。
// receiver_id 为收通知者，actor_id 为触发者（actor==receiver 的通知在 service 层被跳过，不入库）。
// target_id 为触发动作所关联内容（帖子/博客）的内部雪花主键，0 表示无具体目标（如关注通知）；
// 对外展示时由 service 把 target_id 反解为对应内容的 public_id（targetPublicId），绝不暴露雪花主键。
type SysNotification struct {
	BaseModel
	// ReceiverID 收通知者用户ID（内部雪花主键）。
	ReceiverID int64 `json:"-" gorm:"column:receiver_id"`
	// ActorID 触发通知者用户ID（内部雪花主键）。
	ActorID int64 `json:"-" gorm:"column:actor_id"`
	// Type 通知类型（follow/comment/like）。
	Type string `json:"type" gorm:"column:type"`
	// TargetType 目标类型（post/blog，关注类为空串）。
	TargetType string `json:"targetType" gorm:"column:target_type"`
	// TargetID 目标内容内部主键（0 表示无目标）。
	TargetID int64 `json:"-" gorm:"column:target_id"`
	// Content 通知摘要文案（如「评论了你的帖子」）。
	Content string `json:"content" gorm:"column:content"`
	// IsRead 是否已读（0 未读 / 1 已读）。
	IsRead int `json:"isRead" gorm:"column:is_read"`
}

// TableName 表名。
func (SysNotification) TableName() string { return "sys_notification" }

File: tide-canvas-frontend/apps/web/types/notification.ts
import type { PageQuery } from "./api";

/** 通知类型：关注 / 评论 / 点赞 / 打赏。 */
export type NotificationType = "follow" | "comment" | "like" | "tip";

/** 通知目标类型：社区帖子 / 博客（关注类为空串）。 */
export type NotificationTargetType = "post" | "blog" | "";

/** 触发通知者的用户摘要。id 为对方 public_id。 */
export interface NotificationActorVO {
  id: string;
  username: string;
  nickname: string;
  avatar: string;
}

/** 通知列表项。 */
export interface NotificationVO {
  /** 通知自身ID（用于按条标记已读，前端仅回传不展示）。后端雪花主键，字符串传输避免精度丢失 */
  id: string;
  /** 触发者用户摘要（账号已删时各字段为空） */
  actor: NotificationActorVO;
  /** 通知类型 */
  type: NotificationType;
  /** 目标类型（post/blog，关注类为空串） */
  targetType: NotificationTargetType;
  /** 关联内容的对外 public_id（由后端反解；无目标或反解不到为空串）。
   *  前端据 targetType 跳转 /community/{targetPublicId} 或 /blogs/{targetPublicId}。 */
  targetPublicId: string;
  /** 通知摘要文案（如「评论了你的帖子」） */
  content: string;
  /** 是否已读 */
  isRead: boolean;
  /** 仅关注类(type=follow)有意义：当前用户是否已回关该 actor（用于「回关 / 已关注」持久态）。非关注类恒 false。 */
  followedByMe: boolean;
  /** 通知时间 */
  createTime: string;
}

/** 通知列表分页查询。type 为可选类型过滤（空表示全部）。 */
export interface NotificationQuery extends PageQuery {
  type?: NotificationType;
}

/** 未读数响应。 */
export interface UnreadCountVO {
  count: number;
}
